lldb/source/Core/ModuleList.cpp

```diff
@@ -310,6 +310,14 @@ ModuleList::ModuleList(const ModuleList &rhs) : m_modules(), m_modules_mutex() {
   m_modules = rhs.m_modules;
 }
 
+ModuleList::ModuleList(ModuleList &&rhs) : m_modules(), m_modules_mutex() {
+  std::lock_guard<std::recursive_mutex> lhs_guard(m_modules_mutex);
```

Comment:
This should not be needed, since nobody can have access to this object while we're still in the constructor.